spring-boot-project/spring-boot-testcontainers/src/main/java/org/springframework/boot/testcontainers/service/connection/activemq/ActiveMQContainerConnectionDetailsFactory.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2012-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.testcontainers.service.connection.activemq;
+
+import java.util.Map;
+
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+
+import org.springframework.boot.autoconfigure.jms.activemq.ActiveMQConnectionDetails;
+import org.springframework.boot.testcontainers.service.connection.ContainerConnectionDetailsFactory;
+import org.springframework.boot.testcontainers.service.connection.ContainerConnectionSource;
+import org.springframework.boot.testcontainers.service.connection.ServiceConnection;
+
+/**
+ * {@link ContainerConnectionDetailsFactory} to create {@link ActiveMQConnectionDetails}
+ * from a {@link ServiceConnection @ServiceConnection}-annotated {@link GenericContainer}
+ * using the {@code "symptoma/activemq"} image.
+ *
+ * @author Eddú Meléndez
+ */
+class ActiveMQContainerConnectionDetailsFactory
+		extends ContainerConnectionDetailsFactory<ActiveMQConnectionDetails, Container<?>> {
+
+	ActiveMQContainerConnectionDetailsFactory() {
+		super("symptoma/activemq");
```

Comment:
I've updated this from `activemq` to `symptoma/activemq` because it's similar to the Testcontainers mechanism to make sure this works with this image or a similar one to this. The image besides running on arm too it provides good mechanism to set username and password